src/org/zaproxy/zap/extension/domxss/TestDomXSS.java

```diff
@@ -574,5 +584,72 @@ public int getWascId() {
         		AlertThreshold.MEDIUM};
     }
 
+	@Override
+	public void scan(HttpMessage msg, NameValuePair originalParam) {
+		if (originalParam.getType() != NameValuePair.TYPE_QUERY_STRING) {
+			return; // Exit if it isn't a URL param
+		}
+		super.scan(msg, originalParam);
+	}
+
+	@Override
+	public void scan(HttpMessage msg, String paramName, String attack) {
+		Stats.incCounter("domxss.scan.count");
+
+		ArrayList<WebDriverWrapper> drivers = new ArrayList<WebDriverWrapper>();
+
+		WebDriverWrapper fxDriver;
+		try {
+			fxDriver = this.getFirefoxDriver();
+		} catch (Exception e) {
+			getLog().warn("Skipping scanner, failed to start Firefox: " + e.getMessage());
+			getParent().pluginSkipped(this, Constant.messages.getString("domxss.skipped.reason.browsererror"));
+			return;
+		}
+
+		drivers.add(fxDriver);
+
+		for (String attackVector : PARAM_ATTACK_STRINGS) {
+			if (scan(drivers, paramName, attackVector)) {
+				if (!Plugin.AlertThreshold.LOW.equals(this.getAlertThreshold())) {
+					// Only report one issue per URL unless
+					// Alert threshold is LOW
+					break;
+				}
+			}
+		}
+		this.returnFirefoxDriver(fxDriver);
+	}
+	
+	public boolean scan(ArrayList<WebDriverWrapper> drivers, String paramName, String attackVector) {
```

Comment:
Your comment about code duplication (https://github.com/zaproxy/zap-extensions/pull/1403#discussion_r172333427) was lost on me....

Reply:
Both methods, `scan(ArrayList<WebDriverWrapper>, String)` and `scan(ArrayList<WebDriverWrapper>, String, String)`, are doing the same things except for the preparation of `currURL`, so if `currURL` is passed as parameter we can use the existing method for both cases.